src/Orleans.Core/Configuration/Options/ClusterMembershipOptions.cs

```diff
@@ -91,6 +91,19 @@ public class ClusterMembershipOptions
         public int NumVotesForDeathDeclaration { get; set; } = DEFAULT_LIVENESS_NUM_VOTES_FOR_DEATH_DECLARATION;
         public const int DEFAULT_LIVENESS_NUM_VOTES_FOR_DEATH_DECLARATION = 2;
 
+        /// <summary>
+        /// To be used with <see cref="CleanupDeadEntriesTimeout"/>
+        /// </summary>
+        public TimeSpan DeleteEntriesOlderThan { get; set; } = DEFAULT_DELETE_ENTRIES_OLDER_THAN;
+        public static readonly TimeSpan DEFAULT_DELETE_ENTRIES_OLDER_THAN = TimeSpan.FromDays(7);
+
+        /// <summary>
+        /// If set, will check if there are older entries than <see cref="DeleteEntriesOlderThan"/> in the membership table,
+        /// and will delete them
+        /// </summary>
+        public TimeSpan? CleanupDeadEntriesTimeout { get; set; } = DEFAULT_DELETE_ENTRIES_OLDER_THAN;
+        public static readonly TimeSpan? DEFAULT_CLEANUP_DEAD_ENTRIES_TIMEOUT = null;
```

Comment:
This was confusing to me before reading the code. I think we should rename it to something like `DefunctSiloCleanupPeriod` and change the comment to:

```xml
/// <summary>
/// The duration between membership table cleanup operations. When this period elapses, all defunct silo
// entries older than <see cref="DefunctSiloExpiration" /> are removed. This value is per-silo.
/// </summary>
```

Thoughts? I'm not set on those names, but 'timeout' to me says that the process will be cancelled after that amount of time.

Reply:
Yes I like your names better